Feat: Implement ME Endpoint

ajimae opened the pull request:
### Summary:
First draft of the ME endpoint implementation.

- [x] anonymous use authentication
- [x] anonymous user checkout
- [x] user password authentication
- [x] user password checkout
- [x] anonymous active cart merge with user active cart
- [x] anonymous user full cart operations (add items, remove items from cart)
- [x] password user full cart operations (add items, remove items from cart)

### Installation
- Clone/Download the example folder
- Create a `.env` file with the following entries
```txt
CTP_CLIENT_ID=
CTP_PROJECT_KEY=
CTP_CLIENT_SECRET=
```
These credentials can be gotten from the commercetools [merchant center](https://mc.europe-west1.gcp.commercetools.com/login?reason=unauthorized&redirectTo=https%3A%2F%2Fmc.europe-west1.gcp.commercetools.com)

- Open up a Terminal window and navigate to `examples/me/server` and run `yarn install` feel free to use `npm` if you wish
- After the installation completes, run `yarn start:dev` within the (same) server directory, this should spin up the server.
- On a separate Terminal window, navigate to `examples/me/client` and run `yarn install` also, feel free to use `npm` if you wish
- Also, once the installation completes, run `yarn start:dev` within the (same) server directory, this will open up a browser Tab with a list of products, assuming everything went well.


### Note:
- For now, the anonymous carts are semi-manually copied over to the user  active cart and removed from the anonymous cart - this is currently the workaround that works.
- Work is still being done to fully automate this process.
- However, for now it works as intended.

jherey commented:
The installation guide should be in the `examples/me/server/README.md` file. Also, `PORT` should be added to the env file.

ajimae commented:
> The installation guide should be in the `examples/me/server/README.md` file. Also, `PORT` should be added to the env file.

I have moved the README.md file to the root of the `me/` directory and updated the README.md file also.

jherey reviewed (approved):
Very good job and it works great 🚀
